### DISCUSSION TITLE
Mixed Content on MAUI Hybrid

### DISCUSSION THREAD

**Discussion by Wolfleader101** [dotnet/maui#20885] [Q&A]
Hey guys,

Has anyone been able to get mixed content working in a Maui Blazor hybrid successfully?

I've seen a few discussions about this before. However, most of them ended in a closed discussion.

What are the chances that mixed content will ever be allowed?

I am working on an app that needs to work on a closed network with no internet connection. On the same closed network, there is a web server running. However, because it's a closed network, I can't use HTTPS and don't have access to domain names, instead, we just connect to the webserver via the IP i.e. `10.1.1.10`.

I have tried everything possible to enable mixed content in the app for Android however nothing works.

What I have tried

MainPage.xaml
```xaml
    <BlazorWebView x:Name="blazorWebView"
                   HostPage="wwwroot/index.html"
                   android:WebView.MixedContentMode="AlwaysAllow"
```

MainPage.cs
```cs
    public partial class MainPage : ContentPage
    {
        public MainPage()
        {
            InitializeComponent();
            blazorWebView.BlazorWebViewInitialized += Bwv_BlazorWebViewInitialized;
        }

        private void Bwv_BlazorWebViewInitialized(object sender, Microsoft.AspNetCore.Components.WebView.BlazorWebViewInitializedEventArgs e)
        {
#if ANDROID
          e.WebView.Settings.MixedContentMode = Android.Webkit.MixedContentHandling.AlwaysAllow;
#endif
        }
    }
```

xml/network_security_config.xml
```xml
<?xml version="1.0" encoding="utf-8"?>
<network-security-config>
  <base-config cleartextTrafficPermitted="true" />
  <domain-config cleartextTrafficPermitted="true">
    <domain includeSubdomains="true">10.2.69.21</domain>
  </domain-config>
</network-security-config>

```

AndroidManifest
```xml
<?xml version="1.0" encoding="utf-8"?>
<manifest xmlns:android="http://schemas.android.com/apk/res/android">
    <application android:allowBackup="true" android:icon="@mipmap/appicon" android:roundIcon="@mipmap/appicon_round" android:supportsRtl="true" android:networkSecurityConfig="@xml/network_security_config" android:usesCleartextTraffic="true"></application>
    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
    <uses-permission android:name="android.permission.INTERNET" />
</manifest>
```

None of these work, as I still get the mixed content generation error.

**Comment by rmnaderdev** [comment]
Also having this issue in Feb 2025. Have not been able to find a workaround for loading content from a API over http into the webview.
